src/prefect/logging/loggers.py

```diff
@@ -205,6 +206,31 @@ def task_run_logger(
     )
 
 
+def get_worker_logger(worker: "BaseWorker", name: Optional[str] = None):
+    """
+    Create a worker logger with the worker's metadata attached.
+
+    If the worker has a backend_id, it will be attached to the log records.
+    If the worker does not have a backend_id a basic logger will be returned.
+    If the worker does not have a backend_id attribute, a basic logger will be returned.
+    """
+
+    worker_log_name = name or f"workers.{worker.__class__.type}.{worker.name.lower()}"
+    try:
+        if worker.worker_id:
+            return PrefectLogAdapter(
+                get_logger(worker_log_name),
+                extra={
+                    "worker_id": str(worker.worker_id),
+                },
+            )
+        else:
+            return get_logger(worker_log_name)
+    except AttributeError:
+        # Going to throw this in there in case if we have a user defined worker w/o a backend_id
+        return get_logger(worker_log_name)
```

Comment:
I have slight preference for using `getattr(worker, "worker_id", None)` rather than catching an `AttributeError`.